### ISSUE TITLE
about {spatial / temporal} position embedding between code and paper

### ISSUE THREAD

**Issue by wenwen12321** [Vegetebird/MHFormer#15]
Hi~ Excellent work and thanks for releasing your code.

I am confused about "Temporal_pos_embed" which is in the "trans.py" file's Transformer part.
In your paper,  Fig3.(b) show that MHG module only has "Spatial Position Embedding," not "Temporal Position Embedding."

Following is the definition of {spatial/temporal} position embedding in PoseFormer; I am not sure which position embedding is precisely you want to use while there are differences between code and paper.  

```
# In PoseFormer "model_poseformer.py" file
self.Spatial_pos_embed = nn.Parameter(torch.zeros(1, num_joints, embed_dim_ratio))
self.Temporal_pos_embed = nn.Parameter(torch.zeros(1, num_frame, embed_dim))

# My confusion is:
# In MHFormer paper, Fig3(b) MHG only has "Spatial pos embed"
# In MHFormer code, "trans.py" file got a "Temporal_pos_embed" in Transformer part.
self.Temporal_pos_embed = nn.Parameter(torch.zeros(1, length, embed_dim))
```

If there is any misunderstanding, please let me know!!
Thank you so much

**Comment by Vegetebird** [comment]
Hi~
This is a mistake of variable name, you can ignore this.

**Comment by wenwen12321** [comment]
Thanks for your prompt reply.

Does a "mistake of variable name" in the code (not paper)?
So I turn "self.Temporal_pos_embed" to "self.Spatial_pos_embed" is correct right?

But in PoseFormer, the input dimension of {Spatial/ Temporal} pos_embed are not the same.
- 【Spatial_pos_embed】     got the parameter **"embed_dim_ratio"** 
- 【Temporal_pos_embed】 got the parameter **"embed_dim"** which is **"num_joints * spatial embedding dim ratio".**

So should I modify the parameter from "embed_dim" to "embed_dim_ratio" when turn "self.Temporal_pos_embed" to "self.Spatial_pos_embed".

Thanks!

**Comment by Vegetebird** [comment]
This is a mistake of variable name in the code, so you can ignore it. You don't need to modify anything and can test with our pretrained model. 
Our "Spatial Position Embedding" is different from PoseFormer,  you can refer our paper or code for more details.

**Comment by wenwen12321** [comment]
I see.
Thanks for your reply~~

**Comment by Vegetebird** [comment]
We have renamed modules in the code and made the code easier to understand. 
You can redownload the repo and the pretrained model.

**Comment by wenwen12321** [comment]
thank you so much!!
